nmostesting/suites/BCP0060101Test.py

```diff
@@ -35,110 +38,454 @@ def __init__(self, apis):
-                        continue
-                    found_video_jxsv = True
+        try:
+            found_video_jxsv = False
 
-                    url = self.node_url + "sources/" + flow["source_id"]
-                    source_valid, source_response = self.do_request("GET", url)
-                    if not source_valid or source_response.status_code != 200:
-                        return test.FAIL("Unexpected response from the Node API: {}".format(source_response))
+            for flow in self.is04_resources["flows"]:
+                if flow["media_type"] != "video/jxsv":
+                    continue
+                found_video_jxsv = True
 
-                    source = source_response.json()
+                source = source_map[flow["source_id"]]
 
-                    if source["format"] != "urn:x-nmos:format:video":
-                        return test.FAIL("Source {} MUST indicate format with value 'urn:x-nmos:format:video'"
-                                         .format(source["id"]))
+                if source["format"] != "urn:x-nmos:format:video":
+                    return test.FAIL("Source {} MUST indicate format with value 'urn:x-nmos:format:video'"
+                                     .format(source["id"]))
 
-                if found_video_jxsv:
-                    return test.PASS()
+            if found_video_jxsv:
+                return test.PASS()
 
-            except json.JSONDecodeError:
-                return test.FAIL("Non-JSON response returned from Node API")
-            except KeyError as e:
-                return test.FAIL("Expected key '{}' not found in response from {}".format(str(e), url))
+        except KeyError as ex:
+            return test.FAIL("Expected attribute not found in IS-04 resource: {}".format(ex))
 
         return test.UNCLEAR("No JPEG XS Flow resources were found on the Node")
+
+    def test_03(self, test):
+        """JPEG XS Senders have the required attributes"""
+
+        reg_api = self.apis[SENDER_REGISTER_KEY]
+
+        for resource_type in ["senders", "flows"]:
+            valid, result = self.get_is04_resources(resource_type)
+            if not valid:
+                return test.FAIL(result)
+
+        flow_map = {flow["id"]: flow for flow in self.is04_resources["flows"]}
+
+        reg_path = reg_api["spec_path"] + "/sender-attributes"
+        reg_schema = load_resolved_schema(reg_path, "sender_register.json", path_prefix=False)
+
+        try:
+            found_video_jxsv = False
+            warn_st2110_22 = False
+            warn_message = ""
+
+            for sender in self.is04_resources["senders"]:
+                if sender["flow_id"] is None:
+                    continue
+                if sender["flow_id"] not in flow_map:
+                    continue
+                flow = flow_map[sender["flow_id"]]
+                if flow["media_type"] != "video/jxsv":
+                    continue
+                found_video_jxsv = True
+
```

Comment:
Yeah, I would be tempted to do something like:

```
jxsv_senders = [s in self.is04_resources["senders"]
                if s["flow_id"]
                and flow_map[s["flow_id"]]
                and flow_map[s["flow_id"]]["media_type"] == "video/jxsv"]

for sender in jxsv_senders:
    ....

```
and then it would avoid the need for the flag as you can use `len(jxsv_sender)` in later checks.